docs/metrics/bleu.md

```diff
@@ -0,0 +1,132 @@
+**Reference Sentence**: Kolena is a comprehensive machine learning testing and debugging platform to surface hidden model behaviors and take the mystery out of model development
+
+??? example "Step 1: Tokenization & n-Grams"
+    Splitting our sentences up into 1-, ..., 4-grams, we get:
+
+    **Generated Sentence**:
+
+    | n | n-Grams |
+    | ---   | ---                        |
+    | 1 | ["kolena", "is", "an", "ml", "testing", "and", "debugging", "platform", "to", "find", "hidden", "model", "behaviors", "and", "demystify", "model", "development"]|
+    | 2 | ["kolena is", "is an", "an ml", "ml testing", "testing and", "and debugging", "debugging platform", "platform to", "to find", "find hidden", "hidden model", "model behaviors", "behaviors and", "and demystify", "demystify model", "model development"]|
+    | 3 | ["kolena is an", "is an ml", "an ml testing", "ml testing and", "testing and debugging", "and debugging platform", "debugging platform to", "platform to find", "to find hidden", "find hidden model", "hidden model behaviors", "model behaviors and", "behaviors and demystify", "and demystify model", "demystify model development"] |
+    | 4 | ["kolena is an ml", "is an ml testing", "an ml testing and", "ml testing and debugging", "testing and debugging platform", "and debugging platform to", "debugging platform to find", "platform to find hidden", "to find hidden model", "find hidden model behaviors", "hidden model behaviors and", "model behaviors and demystify", "behaviors and demystify model", "and demystify model development"]  |
+
+    **Reference Sentence**:
+
+    | n | n-Grams |
+    | ---   | ---                        |
+    | 1 | ["kolena", "is", "a", "comprehensive", "machine", "learning", "testing", "and", "debugging", "platform", "to", "surface", "hidden", "model", "behaviors", "and", "take", "the", "mystery", "out"]|
+    |...|
+    | 4 | ["kolena is a comprehensive", "is a comprehensive machine", "a comprehensive machine learning", "comprehensive machine learning testing", "machine learning testing and", "learning testing and debugging", "testing and debugging platform", "and debugging platform to", "debugging platform to surface", "platform to surface hidden", "to surface hidden model", "surface hidden model behaviors", "hidden model behaviors and", "model behaviors and take", "behaviors and take the", "and take the mystery", "take the mystery out", "the mystery out of", "mystery out of model", "out of model development"]  |
+
+??? example "Step 2: Calculate n-Gram Overlap"
+    Next, lets calculate the clipped precision scores for each of the n-Grams. Recall that the precision formula is:
+
+    $$
+    p_i = \frac{\text{Clipped} \text{ count of matching i-grams in machine-generated text}^1}{\text{Total number of i-grams in machine-generated text}}
+    $$
+
+    <center>
+
+    | n | Clipped Precision  |
```

Comment:
I ran the same example on [hugging face](https://huggingface.co/spaces/evaluate-metric/bleu) and I got slight different numbers for precision. It could be because of the tokenizer or any pre-processing on the sentences but it would be nice if you can double check the numbers.

```
{'bleu': 0.2941655597471847, 'precisions': [0.7647058823529411, 0.5625, 0.3333333333333333, 0.21428571428571427], 'brevity_penalty': 0.7026185226629954, 'length_ratio': 0.7391304347826086, 'translation_length': 17, 'reference_length': 23}
```

Reply:
Interesting, I used [this repo](https://github.com/neural-dialogue-metrics/BLEU) to calculate the clipped precision scores, but I think it's using a different definition of clipped precision. I'll use the HuggingFace defn instead since it's more popular.